From dcde0ae74c2a95c2f65bff1bb897481609ef1b00 Mon Sep 17 00:00:00 2001
From: Nicola Fontana <ntd@entidi.it>
Date: Tue, 21 Jul 2026 08:09:34 +0200
Subject: [PATCH] Fix #227: support C++20

`gcc 16` changed its default C++ standard from 17 to 20, and in that
standard the TChar * overload of the extractor operator has been dropped
because too dangerous (no explicit size check was performed).

Explicitly use the `read` method instead.
---
 tool/DataTypeHandler.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tool/DataTypeHandler.cpp b/tool/DataTypeHandler.cpp
index 9f0ebc62..fc47ed22 100644
--- a/tool/DataTypeHandler.cpp
+++ b/tool/DataTypeHandler.cpp
@@ -184,7 +184,7 @@ size_t DataTypeHandler::interpretAsType(
                     << dataSize << " > " << targetSize << ")";
                 throw SizeException(err.str());
             }
-            str >> (char *) target;
+            str.read((char *) target, targetSize);
             break;
         case 0x0011: // double
             {
-- 
2.55.0

